packages/forestadmin-client/src/permissions/action-permission.ts

```diff
@@ -5,15 +5,20 @@ import generateActionsFromPermissions, {
   ActionPermissions,
 } from './generate-actions-from-permissions';
 import { ForestAdminServerInterface } from '../types';
+import TTLCache from '../utils/ttl-cache';
 
 export default class ActionPermissionService {
-  private permissionsPromise: Promise<ActionPermissions> | undefined;
-  private permissionExpirationTimestamp: number | undefined;
+  private permissionsCache: TTLCache<ActionPermissions>;
 
   constructor(
     private readonly options: ForestAdminClientOptionsWithDefaults,
     private readonly forestAdminServerInterface: ForestAdminServerInterface,
-  ) {}
+  ) {
+    this.permissionsCache = new TTLCache(
+      this.fetchEnvironmentPermissions.bind(this),
```

Comment:
```suggestion
      (_key: string) => this.fetchEnvironmentPermissions(),
```
Dev Xp prefers an anonymous function instead of a bind. 🤷‍♂️